### PR TITLE
[INTERNAL] /transaction/{transactionhash} returns TSSResponse

### LINKED ISSUE #100: Make the tss endpoint transaction/{hash} return the same response as the webhook
Right now, the /transaction/{transactionhash} endpoint returns a GetTransactionResponse: https://github.com/stellar/wallet-backend/blob/main/internal/serve/httphandler/tss_handler.go#L133

Change this so that it returns the same response as the webhook: https://github.com/stellar/wallet-backend/blob/main/internal/tss/types.go#L185

You'll have to look up the latest try for that transaction to get its error status code

### PR DESCRIPTION
**Pull request by aristidesstaffieri**
Closes #100 

### What
Makes `GetTransaction` return a `TSSResponse` to match the webhook response.

### Why
To move towards a unified response type.

### Known limitations
N/A

### Issue that this PR addresses
https://github.com/stellar/wallet-backend/issues/100

#### PR Structure

- [x] It is not possible to break this PR down into smaller PRs.
- [x] This PR does not mix refactoring changes with feature changes.
- [x] This PR's title starts with name of package that is most changed in the PR, or `all` if the changes are broad or impact many packages.

#### Thoroughness

- [x] This PR adds tests for the new functionality or fixes.
- [x] All updated queries have been tested (refer to [this](https://stackoverflow.com/a/29163465) check if the data set returned by the updated query is expected to be same as the original one).

#### Release

- [ ] This is not a breaking change.
- [x] This is ready to be tested in development.
- [x] The new functionality is gated with a feature flag if this is not ready for production.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ you can set it to any value. For production, you should set it to a secure passp
 3. Start the containers:
 
 ```bash
-docker-compose up
+docker compose up
```
Comment (gouthamp-stellar) [inline README.md:67]: both these commands work, so there's no need to really change it

Comment (aristidesstaffieri) [reply README.md:67]: oh weird yeah you're right, not sure why I thought the previous syntax was deprecated. Will put it back.

File: internal/serve/httphandler/tss_handler.go [hunk 2]
```diff
@@ -155,9 +156,18 @@ func (t *TSSHandler) GetTransaction(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		httperror.InternalServerError(ctx, "unable to get tx try "+tx.Hash, err, nil, t.AppTracker).Render(w)
+		return
+	}
+
+	httpjson.Render(w, tss.TSSResponse{
+		TransactionHash:       tx.Hash,
+		TransactionResultCode: fmt.Sprint(tssTry.Code),
+		Status:                tx.Status,
+		CreatedAt:             tx.CreatedAt.Unix(),
```
Comment (gouthamp-stellar) [inline internal/serve/httphandler/tss_handler.go:169]: this you should be fetching from the try, not the transaction. So its try.CreatedAt.Unix(). The tx.CreatedAt just tells us when the transaction was submitted to TSS by a client, whereas try.CreatedAt tells us when a transaction was actually created by the network

Comment (aristidesstaffieri) [reply internal/serve/httphandler/tss_handler.go:169]: ah got it, thanks for catching it. Patch incoming.

File: README.md (hunk 1 above)
Comment (JakeUrban) [reply README.md:67]: The previous syntax is [deprecated](https://docs.docker.com/compose/releases/migrate/), isn't it?

File: internal/serve/httphandler/tss_handler_test.go [hunk 3]
```diff
@@ -268,12 +268,13 @@ func TestGetTransaction(t *testing.T) {
 		resp := rw.Result()
 		respBody, err := io.ReadAll(resp.Body)
 		require.NoError(t, err)
-		var getTxResp GetTransactionResponse
+		var getTxResp tss.TSSResponse
 		_ = json.Unmarshal(respBody, &getTxResp)
+		fmt.Printf("%+v\n", getTxResp)
```
Comment (JakeUrban) [inline internal/serve/httphandler/tss_handler_test.go:273]: This should be removed right?

Comment (aristidesstaffieri) [reply internal/serve/httphandler/tss_handler_test.go:273]: whoops yes debug log I left in. Thanks!

File: internal/serve/httphandler/tss_handler_test.go [hunk 4]
```diff
@@ -290,12 +291,12 @@ func TestGetTransaction(t *testing.T) {
...
 		require.NoError(t, err)
-		var getTxResp GetTransactionResponse
+		var getTxResp tss.TSSResponse
 		_ = json.Unmarshal(respBody, &getTxResp)
 
 		assert.Equal(t, http.StatusNotFound, resp.StatusCode)
-		assert.Empty(t, getTxResp.Hash)
-		assert.Empty(t, getTxResp.XDR)
+		assert.Empty(t, getTxResp.TransactionHash)
+		assert.Empty(t, getTxResp.EnvelopeXDR)
```
Comment (gouthamp-stellar) [inline internal/serve/httphandler/tss_handler_test.go:299]: It would be good to add a unit test for when GetLatestTry returns an empty result, which means EnvelopeXDR and resultXDR will be empty

Comment (gouthamp-stellar) [inline internal/serve/httphandler/tss_handler_test.go:299]: can we also assert for resultXDR here?

File: internal/serve/httphandler/tss_handler_test.go [hunk 5]
```diff
@@ -290,12 +291,12 @@ func TestGetTransaction(t *testing.T) {
 		resp := rw.Result()
 		respBody, err := io.ReadAll(resp.Body)
 		require.NoError(t, err)
-		var getTxResp GetTransactionResponse
+		var getTxResp tss.TSSResponse
```
Comment (gouthamp-stellar) [inline internal/serve/httphandler/tss_handler_test.go:294]: nit: rename this vat tssResp

**Review by gouthamp-stellar — CHANGES_REQUESTED** [review]
a few small changes to be made and this should be good to go

File: README.md (hunk 1 above)
Comment (gouthamp-stellar) [reply README.md:67]: ah, fair enough, you can leave this change in here then

File: internal/serve/httphandler/tss_handler.go (hunk 2 above)
Comment (aristidesstaffieri) [reply internal/serve/httphandler/tss_handler.go]: done in 71dd0315064888d82e67146b055ca0a49b9cc9c5

File: internal/serve/httphandler/tss_handler_test.go (hunk 3 above)
Comment (aristidesstaffieri) [reply internal/serve/httphandler/tss_handler_test.go]: done in 73a9b82ef79ea1c888c3ac93fcf21f8b9b8ffa4f

File: internal/serve/httphandler/tss_handler_test.go (hunk 5 above)
Comment (aristidesstaffieri) [reply internal/serve/httphandler/tss_handler_test.go]: done in 3177c18dc88032c32c5fc21d0b99230248077203

File: internal/serve/httphandler/tss_handler_test.go (hunk 4 above)
Comment (aristidesstaffieri) [reply internal/serve/httphandler/tss_handler_test.go]: done in 1c80ae8d2f8bf2c909a54eab52d8bfe33ac21e34

Comment (aristidesstaffieri) [reply internal/serve/httphandler/tss_handler_test.go]: done in 7d2d1c0b21d2df908d377247bed0ba835ab43231

**Comment by aristidesstaffieri** [comment]
> a few small changes to be made and this should be good to go

Thanks for the review. I've added all the requested changes.

**Review by gouthamp-stellar — APPROVED** [review]
